predicators/datasets/generate_atom_trajs_with_vlm.py

```diff
@@ -735,6 +739,178 @@ def _query_vlm_to_generate_ground_atoms_trajs(
     return ground_atoms_trajs
 
 
+def _generate_ground_atoms_trajs_from_synthesized_predicates(
+        image_option_trajs: List[ImageOptionTrajectory], env: BaseEnv,
+        train_tasks: List[Task], known_predicates: Set[Predicate],
+        all_task_objs: Set[Object],
+        vlm: VisionLanguageModel) -> List[List[Set[GroundAtom]]]:
+    """Given a collection of ImageOptionTrajectories, query a VLM to generate
+    predicates over the object features, then generate a ground atom
+    trajectory."""
+    del all_task_objs  # Unused.
+    prompt_dir = "predicators/datasets/vlm_input_data_prompts/vision_api/"
+    candidates = set()
+    for io_traj in image_option_trajs:
+        # 1. Create a prompt based on the image option trajectory.
+        prompt, imgs = _create_prompt_from_image_option_traj(io_traj, env)
+
+        # 2. Query the VLM to propose predicates.
+        response = vlm.sample_completions(prompt,
+                                          imgs,
+                                          0.0,
+                                          CFG.seed,
+                                          num_completions=1)[0]
+        response_file = prompt_dir + "response.txt"
+        with open(response_file, 'w', encoding="utf-8") as f:
+            f.write(response)
+        # 3. Parse the responses into a set of predicates
+        candidates |= _parse_predicate_proposals(response_file, train_tasks,
```

Comment:
hold on - why are you writing the response to a file like this? Don't we already cache responses by default?

Reply:
It was just for convenience. I'll remove it.